datajoint/blob.py

```diff
@@ -487,17 +487,24 @@ def pack_cell_array(self, array):
     def read_datetime(self):
         """deserialize datetime.date, .time, or .datetime"""
         date, time = self.read_value("int32"), self.read_value("int64")
+        date_str = str(date)
+        time_str = str(time)
+
         date = (
-            datetime.date(year=date // 10000, month=(date // 100) % 100, day=date % 100)
+            datetime.date(
+                year=int(date_str[:4]) if date_str[:4] != "" else 0,
+                month=int(date_str[4:6]) if date_str[4:6] != "" else 0,
+                day=int(date_str[-2:] if date_str[-2:] != "" else 0),
+            )
             if date >= 0
             else None
         )
         time = (
             datetime.time(
-                hour=(time // 10000000000) % 100,
```

Comment:
Why is this better? Here is the timing test:

![image](https://user-images.githubusercontent.com/1096169/167879481-36eca5ba-eeee-4782-b2c7-69788a7519b3.png)

Reply:
> The string parsing takes longer according to `%%timeit` tests and is more verbose.

The way that I found that this results in a speedup is I profiled the unpacking of a nparray of 100000 datetime objects and then I overloaded the read_datetime method and profiled unpack again. When looking at the cProfile results the string process method resulted in less total time spent in read_datetime. But it seems like when you profile it you get different results. 

Do you want to tag-up on this sometime? Also I have a python script that generates the cProfile performance profiling which you can then visualize using a python package called snakeviz, I can send you this script and we can look at it to see if there is some error in my logic.

I will attach pictures of the visualized performance profiles below.

before overload:
![image](https://user-images.githubusercontent.com/49455164/167886952-03ac3bb5-12b0-485f-bf9b-67670ad8119c.png)
![image](https://user-images.githubusercontent.com/49455164/167887064-35e77b1b-a385-45ad-8c3c-e4b253a76470.png)

after:
![image](https://user-images.githubusercontent.com/49455164/167887177-bdc6d49c-45f1-4f15-8fac-aba3d41e6140.png)
![image](https://user-images.githubusercontent.com/49455164/167887250-ac57540d-1e42-43f9-bcbb-25fe79d572a9.png)

specifically I looked at the decrease in total execution time as well as the `totime` column which represents total time spent in each method across all method calls. Please let me know what you think @dimitri-yatsenko

Reply:
I just don't see a compelling reason why the string processing would produce a speedup. I think the reason the blob deserialization is slow is because of python's need to loop through the numbers and calling `datetime.date` separately for each time and each date.

Reply:
A real speedup can potentially be produced by using `numpy.datetime64` type support.

Reply:
Here is a 500x improvement in decoding speed:
![image](https://user-images.githubusercontent.com/1096169/167911084-a43a3252-d8c5-4127-8b3a-092e4df36a05.png)

Reply:
`np.datetime64` did not exist when we made the original time serializer.

Reply:
For now, we can recommend the workaround for storing datetimes as `int64` as shown. We can add native support for `datetime64` data type, which would eliminate the need for converting into uint64 and back.